_posts/2022-09-22-cockpit-277.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Cockpit 277
+author: garrett
+date: '2022-09-22'
+tags: cockpit, machines, podman
+slug: cockpit-277
+category: release
+summary: ''
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 277, cockpit-machines 275, and cockpit-podman 54:
+
+
+## login: Improve password manager compatibility
+
+The HTML of the login page has been adjusted to be more compatible with password managers in popular browsers. Usernames and passwords are more likely to be prefilled or selectable, depending on the password manager and browser.
+
+_Thanks to [Jacek Tomasiak](https://github.com/skazi0) for pointing this out and providing a fix!_
+
+## login: Adjust selectors when checking for `:is()` and `:where()` support
+
+Newer, upcoming browser versions have improved next-level support for `:is()` and `:where()` selectors. Cockpit was checking for support an empty usage which passed on browsers in the earlier (current as of writing) implementation. However, browsers have recently updated their parsing support for "[Forgiving Selector Parsing](https://developer.mozilla.org/en-US/docs/Web/CSS/:is#forgiving_selector_parsing)", which caused the newer development versions of Firefox, Chrome, and WebKit to fail this check, preventing the browsers from logging into Cockpit.
+
+The check has now been adjusted so current and upcoming browser versions all pass.
+
+Additionally, hotfixes for older supported affected versions of Cockpit have been published for CentOS, Debian, and Ubuntu. If you have an error while trying to log in with a new browser on an older Cockpit version, please upgrade your version of Cockpit.
```

Comment:
This is a bit more complicated:

  - CentOS 8/9 stream are fixed.
  - RHEL 8.6/9.0 are affected, fixes not yet published. Older RHEL versions are *not* affected.
  - Debian got fixed
  - Ubuntu Kinetic (current devel series) and the backport for 22.04 LTS got fixed. The released version in 22.04 LTS is still affected, and will be fixed (I didn't start the process yet).

Reply:
Thanks! I think we should specify it in the notes.

People who are having this problem would like to know if they're affected, I think.

Reply:
```suggestion
Additionally, hotfixes for older versions of Cockpit have been published for various distributions. If you have an error while trying to log in with a new browser on an older Cockpit version, please upgrade your version of Cockpit.

Updates have been issued for the following distributions and versions:

  - ✅ **CentOS Stream 8 & 9**: Fixed.
  - ✅ **Fedora**: Fixed.
  - ⌛ **RHEL 8.6 & 9.0** are affected, fixes are not yet published.
  - ✅ RHEL versions older than 8.6 are *not* affected.
  - ✅ **Debian**: Fixed.
  - ✅ **Ubuntu Kinetic** (current devel series) and the backport for 22.04 LTS got fixed. 
  - ⌛ **Ubuntu 22.04 LTS** is still affected, but will be fixed soon.
```